### DISCUSSION TITLE
[Improvement]: byte patching help?

### DISCUSSION THREAD

**Discussion by WHots** [YimMenu/YimMenu#2165] [Q&A]
### Which feature

Hey, so not really catching how to use the byte patching methods you have associated with this project.

```
inline void clear_ped_tasks_immediatly(Ped ped)
{

		char pat[] = "\x48\x89\x5C\x24\x00\x57\x48\x83\xEC\x30\xE8\x00\x00\x00\x00\x48\x8B\xF8\x48\x85\xC0\x0F\x84\x00\x00\x00\x00\x8A\x80\x00\x00\x00\x00\xA8\x01\x74\x2F\xA8\x02";
		char msk[] = "xxxx?xxxxxx????xxxxxxxx????xx????xxxxxx";
		
		PVOID address = FindPattern(NULL, sizeof(pat), pat, msk);
		
		memory::byte_patch patch(address, 0xEB);
		patch.apply();
		

		fpClearPedTasksImmedidatly ptf = reinterpret_cast<fpClearPedTasksImmedidatly>(address);
		ptf(ped);

		patch.restore();
}
```

for this **memory::byte_patch patch(address, 0xEB);** I'm obviously getting the param1 wrong because it's not **uint8_t***, regardless of that when i do convert it, by any means at all i get an error saying it's inaccessible.

This is just boilerplate code to kinda show what im doing.

Thanks.

### Reason

n/a

**Comment by WHots** [comment]
@gir489returns - Yeah i get that, but in my particular scenario it's crashing when i use the method.

```
struct task_managed_patches
{
	inline static memory::byte_patch* m_can_clear_task;
};
```

```
inline void clear_ped_tasks_immediatly(Ped ped)
{

	char pat[] = "\x48\x89\x5C\x24\x00\x57\x48\x83\xEC\x30\xE8\x00\x00\x00\x00\x48\x8B\xF8\x48\x85\xC0\x0F\x84\x00\x00\x00\x00\x8A\x80\x00\x00\x00\x00\xA8\x01\x74\x2F\xA8\x02";
	char msk[] = "xxxx?xxxxxx????xxxxxxxx????xx????xxxxxx";
		
	PVOID address = FindPattern(NULL, sizeof(pat), pat, msk);
	g_pointers->m_gta.m_stop_ped_task_immedeatly = address;

	task_managed_patches::m_can_clear_task = memory::byte_patch::make(g_pointers->m_gta.m_stop_ped_task_immedeatly.add(2).as<uint8_t*>(), std::vector{0x90, 0x90, 0x90, 0x90, 0x90}).get(); 
	task_managed_patches::m_can_clear_task->apply();

	fpClearPedTasksImmedidatly ptf = reinterpret_cast<fpClearPedTasksImmedidatly>(address);
	ptf(ped);

	task_managed_patches::m_can_clear_task->restore();
}
```

I'd assume it's the **task_managed_patches::m_can_clear_task = memory::byte_patch::make(g_pointers->m_gta.m_stop_ped_task_immedeatly.add(2).as<uint8_t*>(), std::vector{0x90, 0x90, 0x90, 0x90, 0x90}).get(); ** line somewhere with the first param, but I'm not sure how  **.add(...)** and **.as<uint8_t*>()** are determined.

It's all kinda pseudo for now until i get something working.

**Comment by FIying-Scotsman** [comment]
> but I'm not sure how .add(...) and .as<uint8_t>() are determined.

.add() is used where your pattern takes you to a location near where you want your final pointer to be, and allows you to essentially move it as needed. For example if your pattern takes you to an address 3 bytes away from where you want - you'd add 3. If you make a pattern 10 bytes away - you'd add 10 etc etc. This isn't needed if your pattern lands on the address you want, and is usually used to allow you to make smaller patterns as some of them can be huge.

The type is determined by the return type - in this case a pointer address.

**Reply by WHots** [reply]
Dope, i appreciate that reply.. ill tinker with it in a bit.

**Comment by WHots** [comment]
```
inline void clear_ped_tasks_immediatly(Ped ped)
	{

		MODULEINFO module_info;
		
		K32GetModuleInformation(GetCurrentProcess(), GetModuleHandle(NULL), &module_info, sizeof(MODULEINFO));

		char pattern[] = "\x48\x89\x5C\x24\x00\x57\x48\x83\xEC\x30\xE8\x00\x00\x00\x00\x48\x8B\xF8\x48\x85\xC0\x0F\x84\x00\x00\x00\x00\x8A\x80\x00\x00\x00\x00\xA8\x01\x74\x2F\xA8\x02";
		char mask[] = "xxxx?xxxxxx????xxxxxxxx????xx????xxxxxx";

		uintptr_t address = FindPattern((char*)module_info.lpBaseOfDll, module_info.SizeOfImage, pattern, mask);

		if (!address)
		{
			g_notification_service->push_error("TEST", "Failed fetching pattern.");
			return;
		}
		
		g_pointers->m_gta.m_stop_ped_task_immedeatly = (PVOID)address;

		task_managed_patches::m_can_clear_task = memory::byte_patch::make(g_pointers->m_gta.m_stop_ped_task_immedeatly.as<uint8_t*>(), 0x7B).get(); 
		task_managed_patches::m_can_clear_task->apply();

		fpClearPedTasksImmedidatly ptf = reinterpret_cast<fpClearPedTasksImmedidatly>(address);

		if (ptf != nullptr)
			ptf(ped);
		else
		{
			g_notification_service->push_error("TEST", "function pinter.");
			return;
		}

		task_managed_patches::m_can_clear_task->restore();
	}
```

So before, i was getting an error on my FindPattern, i've adjusted it and now  it's fine? no fookn clue cause now the game just crashes when this method runs, no dialog / notifications.. i placed a little output above **K32GetModuleInformation** like **g_notification_service->push_error("TEST", "function pinter.");** but nothing is displayed... soooo, idek. The pattern is fine and tested in IDA, i've done similar stuff for other games that actual have anti-cheats on my own with my own stuff.... but i'm assuming im using the built in nigga rigged "byte_patch" wrong? cause tbh, just diving into it.. kinda messy and hard to follow, like i essentially mimicked how it's used for blame explosions.

**Comment by maybegreat48** [comment]
Use `g_pointers->m_gta.m_clear_ped_tasks_network` instead
